src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -60,210 +69,280 @@ private void captureOs() {
-            Utils.envVariable("CI_JOB_NAME", providers).ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI job", value));
-            Utils.envVariable("CI_JOB_STAGE", providers).ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI stage", value));
-        }
+            if(CiUtils.isTeamCity(providers)) {
+                Optional<String> teamcityBuildPropertiesFile = Utils.envVariable("TEAMCITY_BUILD_PROPERTIES_FILE", providers);
+                if(teamcityBuildPropertiesFile.isPresent()){
+                    Properties buildProperties = Utils.readPropertiesFile(teamcityBuildPropertiesFile.get(), providers, projectDirectory.get());
+
+                    String teamcityConfigFile = buildProperties.getProperty("teamcity.configuration.properties.file");
+                    if(Utils.isNotEmpty(teamcityConfigFile)) {
+                        Properties configProperties = Utils.readPropertiesFile(teamcityConfigFile, providers, projectDirectory.get());
+
+                        String teamCityServerUrl = configProperties.getProperty("teamcity.serverUrl");
+                        String teamCityBuildId = buildProperties.getProperty("teamcity.build.id");
+                        if (Utils.isNotEmpty(teamCityServerUrl) && Utils.isNotEmpty(teamCityBuildId)) {
+                            String buildUrl = appendIfMissing(teamCityServerUrl, "/") + "viewLog.html?buildId=" + urlEncode(teamCityBuildId);
+                            buildScan.link("TeamCity build", buildUrl);
+                        }
+                    }
 
-        if (CiUtils.isTravis(providers)) {
-            Utils.envVariable("TRAVIS_BUILD_WEB_URL", providers).ifPresent(url ->
-                buildScan.link("Travis build", url));
-            Utils.envVariable("TRAVIS_BUILD_NUMBER", providers).ifPresent(value ->
-                buildScan.value("CI build number", value));
-            Utils.envVariable("TRAVIS_JOB_NAME", providers).ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI job", value));
-            Utils.envVariable("TRAVIS_EVENT_TYPE", providers).ifPresent(buildScan::tag);
-        }
+                    String teamCityBuildNumber = buildProperties.getProperty("build.number");
+                    if (Utils.isNotEmpty(teamCityBuildNumber)) {
+                        buildScan.value("CI build number", teamCityBuildNumber);
+                    }
+                    String teamCityBuildTypeId = buildProperties.getProperty("teamcity.buildType.id");
+                    if (Utils.isNotEmpty(teamCityBuildTypeId)) {
+                        buildScan.value("CI build config", teamCityBuildTypeId);
+                    }
+                    String teamCityAgentName = buildProperties.getProperty("agent.name");
+                    if (Utils.isNotEmpty(teamCityAgentName)) {
+                        buildScan.value("CI agent", teamCityAgentName);
```

Comment:
before we were calling the `customValueSearchLinker` why did we change it here? Is it not needed?

Reply:
Well spotted, this one just slipped through the cracks while refactoring, thanks!